pkg/prometheus/operator.go

```diff
@@ -1877,11 +1580,17 @@ func (c *Operator) createOrUpdateTLSAssetSecret(p *monitoringv1.Prometheus) erro
 		Data: map[string][]byte{},
 	}
 
-	for key, asset := range tlsAssets {
-		tlsAssetsSecret.Data[key] = []byte(asset)
+	for i, rw := range p.Spec.RemoteWrite {
+		if err := store.addTLSConfig(p.GetNamespace(), rw.TLSConfig); err != nil {
+			return errors.Wrapf(err, "remote write %d", i)
+		}
+	}
+
+	for key, asset := range store.tlsAssets {
+		tlsAssetsSecret.Data[key.String()] = []byte(asset)
 	}
 
-	_, err = sClient.Get(context.TODO(), tlsAssetsSecret.Name, metav1.GetOptions{})
+	_, err := sClient.Get(context.TODO(), tlsAssetsSecret.Name, metav1.GetOptions{})
```

Comment:
As we add additional parameters to the function body, maybe let's start adding context here too? Not critical here, we can do this as a follow-up.